accounts-db/src/tiered_storage/meta.rs

```diff
@@ -113,6 +116,21 @@ pub struct AccountMetaOptionalFields {
 }
 
 impl AccountMetaOptionalFields {
+    /// Create a new AccountMetaOptionalFields instance with the specified
+    /// fields.  If the value of a field equals to the default, then None
+    /// will be used in that field.
+    pub fn new_from_fields(
```

Comment:
The caller is write_accounts(), where the input is `StorableAccountsWithHashesAndWriteVersions` that doesn't have the concept of Some / None so far.

So the possible locations are:

1. Inside `AccountMetaOptionalFields` where it has the potential of handling all :optional" stuff.
2. Inside write_accounts, but it would also be great to avoid having too much stuff inside one function.
3. Inside StorableAccountsWithHashesAndWriteVersions, which I feel could be the best place.  A dedicated PR is needed for this that will apply the change to AppendVec as well.

Reply:
Removed from this PR and will have this under write_accounts().